src/transformers/models/vibevoice_semantic_tokenizer/modular_vibevoice_semantic_tokenizer.py

```diff
@@ -0,0 +1,386 @@
+
+    For reference, original 2D `ConvNextLayer`:
+    https://github.com/huggingface/transformers/blob/e20df45bf676d80bdddb9757eeeafe6c0c81ecfa/src/transformers/models/convnext/modeling_convnext.py#L120
+    """
+
+    def __init__(self, config, hidden_size, drop_path=0.0, dilation=1, stride=1):
+        super().__init__()
+
+        self.norm = VibeVoiceRMSNorm(hidden_size, eps=config.rms_norm_eps)
+        self.ffn_norm = VibeVoiceRMSNorm(hidden_size, eps=config.rms_norm_eps)
+        self.ffn = VibeVoiceEncoderFeedForward(config, hidden_size)
+        self.gamma = nn.Parameter(config.layer_scale_init_value * torch.ones(hidden_size), requires_grad=True)
+        self.ffn_gamma = nn.Parameter(config.layer_scale_init_value * torch.ones(hidden_size), requires_grad=True)
+
+        # TODO (ebezzam) original code has option for DropPath but is never actually used (and `nn.modules.DropPath` does not exist):
+        # https://github.com/pengzhiliang/transformers/blob/6e6e60fb95ca908feb0b039483adcc009809f579/src/transformers/models/vibevoice/modular_vibevoice_tokenizer.py#L637
+        # however, could be interesting feature for future versions of `ConvNext1dLayer` as the 2D version has it:
+        # https://github.com/huggingface/transformers/blob/e20df45bf676d80bdddb9757eeeafe6c0c81ecfa/src/transformers/models/convnext/modeling_convnext.py#L146
+        if drop_path > 0.0:
+            # possible implementation (that may needed to be adapted for 1D):
+            # https://github.com/huggingface/transformers/blob/e20df45bf676d80bdddb9757eeeafe6c0c81ecfa/src/transformers/models/convnext/modeling_convnext.py#L40
+            raise NotImplementedError("DropPath is not implemented.")
+        self.drop_path = nn.Identity()
+
+        self.mixer = VibeVoiceCausalConv1d(
+            in_channels=hidden_size,
+            out_channels=hidden_size,
+            kernel_size=config.kernel_size,
+            groups=hidden_size,
+            bias=config.bias,
+            dilation=dilation,
+            stride=stride,
+        )
+
+    def forward(self, hidden_states):
+        residual = hidden_states
+        hidden_states = self.norm(hidden_states.transpose(1, 2)).transpose(1, 2)
+        hidden_states = self.mixer(hidden_states)
+        hidden_states = hidden_states * self.gamma.unsqueeze(-1)
+        # (ebezzam) original code (https://github.com/pengzhiliang/transformers/blob/6e6e60fb95ca908feb0b039483adcc009809f579/src/transformers/models/vibevoice/modular_vibevoice_tokenizer.py#L653)
+        # as mentioned above, drop_path is not used and the VibeVoice authors don't use the `forward` method but a custom
+        # call which does `residual + hidden_states` directly (see link below), which is same as using identity
+        # https://github.com/pengzhiliang/transformers/blob/6e6e60fb95ca908feb0b039483adcc009809f579/src/transformers/models/vibevoice/modular_vibevoice_tokenizer.py#L768
+        hidden_states = residual + self.drop_path(hidden_states)
```

Comment:
If we remove drop_path (just Identity at the moment), it could be simplified to just:
```python
hidden_states = residual + hidden_states
```